app/apps/ood_app.rb

```diff
@@ -40,6 +42,29 @@ def url
     end
   end
 
+  # the problem is we need the context :-P
+  def links
+    if role == "files"
+      # assumes Home Directory is primary...
+      [Link.new("Home Directory", OodAppkit.files.url(path: Dir.home), 'home')] + OodFilesApp.new.favorite_paths.map do |path|
+        Link.new(path.to_s, OodAppkit.files.url(path: path), "folder")
+      end
+    elsif role == "shell"
+      if ApplicationController.helpers.login_clusters.count == 0
+        [Link.new("Shell Access", OodAppkit.shell.url,"terminal")]
+      else
+        ApplicationController.helpers.login_clusters.map { |c| Link.new("#{c.metadata.title} Shell Access", OodAppkit.shell.url(host: c.login.host), "terminal") }
+      end
+    elsif role == "batch_connect"
+      batch_connect.sub_app_list.select(&:valid?).map do |sub_app|
+        Link.new(sub_app.title, Rails.application.routes.url_helpers.new_batch_connect_session_context_path(token: sub_app.token))
+      end
+    else
+      # normal, use default icon
+      [Link.new(title, Rails.application.routes.url_helpers.app_path(name, type, owner))]
```

Comment:
This should probably be:

```ruby
[Link.new(title, url)]
```

otherwise it won't read the url in the manifest.yml file for apps other than batch connect apps.